.github/workflows/pathogen-repo-build.yaml

```diff
@@ -199,6 +197,21 @@ jobs:
           | "$NEXTSTRAIN_GITHUB_DIR"/bin/json-to-envvars
           | tee -a "$GITHUB_ENV"
 
+      - if: inputs.runtime == 'aws-batch'
+        uses: aws-actions/configure-aws-credentials@v2
+        with:
+          aws-region: us-east-1
+          aws-access-key-id: ${{ secrets.AWS_ACCESS_KEY_ID }}
+          aws-secret-access-key: ${{ secrets.AWS_SECRET_ACCESS_KEY }}
+          role-to-assume: ${{ secrets.AWS_ACCESS_KEY_ID == '' && 'arn:aws:iam::827581582529:role/GitHubActionsRoleNextstrainBatchJobs' || '' }}
```

Comment:
Could you clarify what's going on here for me?

The `role-to-assume` line seems like it's assuming a role _if_ keys are not set as secrets. Seems sensible, as it allows users to provide their own IAM keys if the role isn't available / suitable to them.

But we also provide the keys here (`with.aws-access-key-id`, `with.aws-secret-access-key`), and from my reading of the [action docs](https://github.com/aws-actions/configure-aws-credentials/blob/main/action.yml) this means we  "want to assume a role using access keys rather than a web identity token". Perhaps if the keys are empty (which they are if we are going to assume a role) then it's as if we didn't provide them and so we'll assume a role via OIDC, but in this case why are they provided to this action at all?

Reply:
> Perhaps if the keys are empty (which they are if we are going to assume a role) then it's as if we didn't provide them and so we'll assume a role via OIDC…

Yes, this is what happens.

> …but in this case why are they provided to this action at all?

Because providing them to this action makes the secrets available in the environment for subsequent steps. This action is the single codepath to making AWS credentials available for the rest of the steps in the job, regardless of how those credentials were obtained.

Reply:
Gotcha. Thanks. [This table](https://github.com/aws-actions/configure-aws-credentials/tree/main#using-this-action) in the action's docs does imply that's what happens, but [the description of the inputs](https://github.com/aws-actions/configure-aws-credentials/blob/main/action.yml#L18-L20) doesn't say anything about that capability.

Reply:
Nod. I don't think GitHub Actions distinguishes between "empty" and "not set/not provided" inputs, so code consuming them doesn't either.